src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java

```diff
@@ -452,6 +456,296 @@ public void testIsNameVisibleTo() {
+                fr261, fq26, null));
+        assertTrue(frLogic.isResponseVisibleForUser(student1.getEmail(), false, student1, studentsEmailInTeam1,
+                fr281, fq28, null));
+        assertFalse(frLogic.isResponseVisibleForUser(instructor2.getEmail(), true, null, null,
+                fr181, fq18, instructor2));
+        assertFalse(frLogic.isResponseVisibleForUser(student7.getEmail(), false, student7, studentsEmailInTeam3,
+                fr271, fq27, null));
+        assertFalse(frLogic.isResponseVisibleForUser(student8.getEmail(), false, student8, studentsEmailInTeam4,
+                fr281, fq28, null));
+        assertFalse(frLogic.isResponseVisibleForUser(student5.getEmail(), false, student5, studentsEmailInTeam2,
+                fr201, fq20, null));
+
+        ______TS("test if visible to giver's team members");
+
+        assertTrue(frLogic.isResponseVisibleForUser(student2.getEmail(), false, student2, studentsEmailInTeam1,
+                fr171, fq17, null));
+        assertTrue(frLogic.isResponseVisibleForUser(student1.getEmail(), false, student1, studentsEmailInTeam1,
+                fr221, fq22, null));
+        assertTrue(frLogic.isResponseVisibleForUser(student7.getEmail(), false, student7, studentsEmailInTeam3,
+                fr251, fq25, null));
+        // other students include giver's team members
+        assertTrue(frLogic.isResponseVisibleForUser(student3.getEmail(), false, student3, studentsEmailInTeam1,
+                fr261, fq26, null));
+        assertFalse(frLogic.isResponseVisibleForUser(student3.getEmail(), false, student3, studentsEmailInTeam1,
+                fr111, fq11, null));
+        assertFalse(frLogic.isResponseVisibleForUser(student4.getEmail(), false, student4, studentsEmailInTeam1,
+                fr123, fq12, null));
+        assertFalse(frLogic.isResponseVisibleForUser(student2.getEmail(), false, student2, studentsEmailInTeam1,
+                fr241, fq24, null));
+
+        ______TS("test if visible to receiver's team members");
+
+        assertTrue(frLogic.isResponseVisibleForUser(student7.getEmail(), false, student7, studentsEmailInTeam3,
+                fr171, fq17, null));
+        assertTrue(frLogic.isResponseVisibleForUser(student1.getEmail(), false, student1, studentsEmailInTeam1,
+                fr231, fq23, null));
+        assertFalse(frLogic.isResponseVisibleForUser(student2.getEmail(), false, student2, studentsEmailInTeam1,
+                fr111, fq11, null));
+        assertFalse(frLogic.isResponseVisibleForUser(student4.getEmail(), false, student4, studentsEmailInTeam1,
+                fr122, fq12, null));
+        assertFalse(frLogic.isResponseVisibleForUser(student6.getEmail(), false, student6, studentsEmailInTeam3,
+                fr271, fq27, null));
+
+        ______TS("test if visible to instructors");
+
+        assertTrue(frLogic.isResponseVisibleForUser(instructor1.getEmail(), true, null, null,
+                fr111, fq11, instructor1));
```

Comment:
I notice that the `isResponseVisibleForUser` method is being called in this test in only two different manners:
```java
// variant 1
frLogic.isResponseVisibleForUser(email, false, student, studentEmails, fr, fq, null);
// variant 2
frLogic.isResponseVisibleForUser(email, true, null, null, fr, fq, instructor);
```
- If all usages (at least the valid ones) are always like that, probably the second boolean parameter is not needed?
- If the answer to the above is no, then there are missing test cases.

Reply:
I believe those are the only two valid cases, at least for valid usages.

Reply:
On a side note, as long as the `isInstructor` field is `true`, it doesn't matter whether `studentsEmailInTeam` is set to `null` or empty set. Using `null` may not be the best option, which I can change later to empty set, because the valid usages are using empty set instead of `null`. This also means the isInstructor is indeed not needed for this method.

Reply:
Do expect one branch is missed in the test, because `studentsEmailInTeam` will never be null.
![image](https://user-images.githubusercontent.com/68739475/169963303-294ce1af-c45d-4214-af28-be1dac017386.png)